apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data/SpannerConnection.cs

```diff
@@ -661,25 +661,44 @@ public async Task WhenSessionPoolReady(CancellationToken cancellationToken = def
         /// <returns>A task which will complete when the session pool has finished shutting down.</returns>
         public async Task ShutdownSessionPoolAsync(CancellationToken cancellationToken = default)
         {
-            DatabaseName databaseName = Builder.DatabaseName;
-            GaxPreconditions.CheckState(databaseName != null, $"{nameof(ShutdownSessionPoolAsync)} cannot be used without a database.");
+            var sessionPoolSegmentKey = GetSessionPoolSegmentKey(nameof(ShutdownSessionPoolAsync));
             await OpenAsync(cancellationToken).ConfigureAwait(false);
-            await _sessionPool.ShutdownPoolAsync(databaseName, cancellationToken).ConfigureAwait(false);
-        }
-
+            await _sessionPool.ShutdownPoolAsync(sessionPoolSegmentKey, cancellationToken).ConfigureAwait(false);
+        }
+
         /// <summary>
         /// Retrieves the database-specific statistics for the session pool associated with the connection string. The connection string must
         /// include a database name.
         /// </summary>
         /// <returns>The session pool statistics, or <c>null</c> if there is no current session pool
         /// for the database specified in the connection string.</returns>
+        [Obsolete("Use GetSessionPoolStatistics instead. Both methods return the same data, but the GetSessionPoolStatistics name better reflects the fact that sessions are pooled on aspects other than database name.")]
         public SessionPool.DatabaseStatistics GetSessionPoolDatabaseStatistics()
         {
             DatabaseName databaseName = Builder.DatabaseName;
             GaxPreconditions.CheckState(databaseName != null, $"{nameof(GetSessionPoolDatabaseStatistics)} cannot be used without a database.");
             return Builder.SessionPoolManager.GetDatabaseStatistics(new SpannerClientCreationOptions(Builder), databaseName);
         }
 
+        /// <summary>
+        /// Retrieves statistics for the session pool associated with the corresponding <see cref="SessionPoolSegmentKey"/>. The connection string must
+        /// include a database name.
+        /// </summary>
+        /// <returns>The session pool statistics, or <c>null</c> if there is no current session pool
+        /// associated to the <see cref="SessionPoolSegmentKey"/>.</returns>

```

Comment:
Nit: associated to => associated with (as in line 684). But we can fix that later easily.